### PR TITLE
Improvised item update (WIP)

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
add: You can now find different types of tape all over maintenance. It's used in making new recipes.
add: You can now make padded, reinforced, and plated jumpsuits with tape and various metals. They're armored, but they slow you down!
add: You can now make a shiv out of some cloth and a glass shard. It doesn't hit as hard, but it's serrated and causes heavy bleeding.
add: You can now make a wrench tomahawk out of a wrench, a glass shard, and some electrical tape.
/:cl:

* Reorganized the display of recipes into something more logical by putting the common ones at the top.
* Added a Clothing category for crafting.
* Added new crafting materials, currently found exclusively around maintenance, such as different types of tape and all-purpose circuit boards.
* Added padded, reinforced, and plated grey jumpsuits. 
  * Padded jumpsuits provide 5% brute and bullet protection, and make you 10% more vulnerable to burn damage. Made from a grey jumpsuit, 3 bedsheets, and 40 office tape.
  * Reinforced jumpsuits provide 10% brute and bullet protection, make you 10% more vulnerable to laser damage, and slow you down by 1. Made from a grey jumpsuit, 2 metal, and 40 electrical tape.
  * Plated jumpsuits provide 20% brute and bullet protection, make you 30% more vulnerable to laser damage, slow you down by 1, and have a unique name. Made from a grey jumpsuit, 2 plasteel, and 40 duct tape.
  * I realize that these may need balancing! If it gets too controversial, I'll make them an armor slot instead so that you can't stack mad armor for days.
* Added the shiv, made from a glass shard and a bedsheet. It deals 4 damage per hit (less than a shard) but causes one tick of heparin bleeding per hit. Every hit has a chance to cause the shiv to regress into a normal shard, with probability increasing by 1% for each hit.

Suggestions welcome!

### CODE REVIEW

**Comment by Anonmare** [comment]
I'd say make shivs AP and capable of fitting in a pocket. They're generally used for surprising and really fucking someone up

**Comment by GunHog** [comment]
I think you goofed something, there are more items added here than you have documented in your OP.

Let us pad our jumpsuits with even more things! Put glass on the outside of a jumpsuit to give it laser and energy protection, but when hit with any type of brute damage, the glass shatters and injures the user!

**Review by Razharas — APPROVED** [review]
Menucrafting is auto-approval from me

**Comment by Xhuis** [comment]
@GunHog Some of the items have yet to be used, but will be put in.

**Comment by Xhuis** [comment]
Thoughts about making the first hit from a shiv have 100% armor pen and like double damage, but it can't be done again after a period of no attacking?

**Comment by Cheridan** [comment]
Make it take a unit of cloth instead of a WHOLE bedsheet, that's ridiculous.
Adding a reagent on stabs seems like a hack.
Only being able to use jumpsuits that are grey for the armor is cheap, dude. Do it right or don't bother.

**Comment by Xhuis** [comment]
* I made it this way because bedsheets are available in perma but cloth is not. Unless you can cut up sheets into cloth now?
* It's the best way to do this. Only other thing is spawns or addtimers, which are blegh.
* I don't know crafting code well enough to give armor values to individual jumpsuits without making a billion recipes, but I'll try.

**Comment by Xhuis** [comment]
Shivs now lodge a shard in the target when they break instead of putting the shard in the attacker's hand. Also added the wrench tomahawk, which serves as a slightly weaker hatchet.

**Comment by Cheridan** [comment]
>I made it this way because bedsheets are available in perma but cloth is not. Unless you can cut up sheets into cloth now?

yes

**Comment by Repukan** [comment]
Perma prisoners should not receive buffs. You're just making sure next time people just get executed instead.

Make them armor slot items, else this is a huge nerf to cults and war ops.

**Comment by Repukan** [comment]
Perma prisoners should not receive buffs. You're just making sure next time people just get executed instead.

Make them armor slot items, else this is a huge nerf to cults and war ops.

**Comment by Razharas** [comment]
To make smart crafting recipes you need to use CheckParts proc, all things have it, its kinda like RefreshParts in machines but is called on the result of crafting, thus you need to make jumpsuits use this proc to upgrade their armour

**Comment by AdamElTablawy** [comment]
If a war op manages to get killed by a shitty naked assistant with a glass shard and some cloth he fucking deserves it

File: code/modules/crafting/craft.dm [hunk 1]
```diff
@@ -1,7 +1,8 @@
 /datum/personal_crafting
 	var/busy
 	var/viewing_category = 1 //typical powergamer starting on the Weapons tab
-	var/list/categories = list(CAT_WEAPON,
+	var/list/categories = list(CAT_CLOTHES,
```
Comment (RemieRichards) [inline code/modules/crafting/craft.dm]: Either don't change the first one, or remove my funny joke.

**Comment by RemieRichards** [comment]
@Xhuis 

>It's the best way to do this. Only other thing is spawns or addtimers, which are blegh.

You mean the system we specifically added, with a fuck load of tuning to get it to that mmmm perfect that's amore state is blegh? I'd prefer to think you don't know how to use it thanks.

**Comment by Xhuis** [comment]
@RemieRichards what.
@Razharas Thanks, I was going to have to make external "jumpsuit upgrade kits", which is eugh.
@Cheridan TIL
Goddamn it's a maintainer party in here

**Comment by Xhuis** [comment]
* Preserved Remie's humor.
* Generalized reinforcement to all jumpsuits. Note that this _overrides_ existing armor values, so you can't stack armor.
* Replaced shiv and pad jumpsuit recipes with cloth instead of bedsheets.

**Comment by RemieRichards** [comment]
armour with higher values should simply fail to upgrade, you shouldn't be able to spend resources to make a piece of armour worse, that's stupid.

File: code/modules/clothing/under/miscellaneous.dm [hunk 2]
```diff
@@ -1,3 +1,34 @@
...
+			U.armor = list(melee = 10, bullet = 10, laser = -10, energy = 0, bomb = 25, bio = 5, rad = 0, fire = 0, acid = 10)
+			slowdown = 0.5 //You're wearing unshaped sheets of metal.
+		if(plasteels >= 2)
+			U.name = "plated [initial(U.name)]"
+			U.desc = "[initial(U.desc)] It's been plated with plasteel, making it quite protective if incredibly heavy."
+			U.armor = list(melee = 20, bullet = 30, laser = -30, energy = 0, bomb = 30, bio = 5, rad = 0, fire = 0, acid = 25)
+			U.slowdown = 1
+		U.forceMove(loc)
+		qdel(src)
+	return
```
Comment (RemieRichards) [inline code/modules/clothing/under/miscellaneous.dm]: trailing return, remove it please.

**Comment by Xhuis** [comment]
I agree with you on that part, but I'm not sure if I can make a recipe fail under certain conditions. I'll keep looking, though.

**Comment by RemieRichards** [comment]
@Xhuis simply have the ```checkParts()``` do it?

**Comment by Xhuis** [comment]
Is CheckParts called during or after the crafting? If during, I'll have to implement a check to make sure it passes, which is odd but should work.

**Comment by RemieRichards** [comment]
after, you'd have to make the ```checkParts()``` fail and then poop out the materials it "used" (which should be inside the clothing so just dump any ```/sheet```` in it) onto the ground

**Comment by Xhuis** [comment]
Looks like it's going well, thank you for the help. One more question if you don't mind: how can I easily check that the old armor values would exceed the new? I can just have it check if armor() is null but then stuff like the CMO jumpsuit that's slightly bio-resistant would fail.

**Comment by RemieRichards** [comment]
@Xhuis the old jumpsuit, the one you're upgrading in the ```CheckParts()```, the one in your ```U``` variable, it's the old jumpsuit, I'm p.sure.

**Comment by Xhuis** [comment]
Yep, it is. I know that `armor` is an assoclist, so I'm trying to find out if I can easily compare the two. If I were to do, say, `armor[1]`, would it check melee in that case? It should be easy enough to just cycle the two lists, if that's the case.

**Comment by RemieRichards** [comment]
yes it would.
looping an assoc list gets you it's keys, indexing it also gets keys, checking:
```
for(var/armourtype in armor)
    if(armor[armouttype] >= proposed_armour[armourtype])
        ... //upgrade
    else
        ... //fuk u
```

**Comment by Xhuis** [comment]
Sounds good enough. One more non code-related thing now, but should I just check the `melee` and `bullet` values and only changing those, so as to keep the same values across the rest?

**Comment by RemieRichards** [comment]
Yes that works (and solves the above)

**Comment by Xhuis** [comment]
Good deal, thanks!

**Comment by xxalpha** [comment]
finally duct tape

**Comment by Xhuis** [comment]
Updates will not be consistent for a while. Big changes coming in my life.

**Comment by xxalpha** [comment]
Are you pregnant

**Comment by Xhuis** [comment]
Yes, and you're the father

**Comment by GunHog** [comment]
H-how do you father a child with a slime? Any fluids would just be absorbed as protein. Aaaaand, Xhuis reproduces by asexual division...Does he just kind of...envelop you, absorb your body, and use the additional mass gained to split himself?

**Comment by Xhuis** [comment]
Slimes eat people's DNA and they split. I just tore open his chest and ate his organs, then split.

**Comment by optimumtact** [comment]
why do you pr pr's that aren't evne finished yet?

I'm gonna start closing these until people PR things that are ready for review

**Comment by Xhuis** [comment]
When I first started, it was drilled into my head primarily by Remie that I should PR things early so code can be reviewed. It's a force of habit now.

**Comment by optimumtact** [comment]
I think systems should be pretty much ready to go before you PR, otherwise if you ask me it's just a waste of time reviewing it because it changes so often

if you want early review just pr to the maintainers repo or something

**Comment by Repukan** [comment]
Maybe Joan can finish it if we ask nicely enough

**Comment by Xhuis** [comment]
This is likely going to be stale for a while anyway, so I'll close it for now.

**Comment by RemieRichards** [comment]
@optimumtact I'd rather it be early enough that we can completely change the implementation if necessary, which tends to be when people don't quite keep up with everything (assoc lists, LAZYmacros, typecaches, addtimer).

If you come in right at the end and say "1 million things are wrong" 9/10 the PR just dies, if you come in early and say "Ok I think you've gone the wrong direction with this, here's how it should be done", when less work as been put it, it's more likely to keep the PR going.

**Pull request closed without merging**